### PR TITLE
Feature/code refactor

### PR DESCRIPTION
**Pull request by tamaragovorcin**

### CODE REVIEW

File: IntegrationWithPharmacies/Controllers/ActionsAndBenefitsController.cs [hunk 1]
```diff
@@ -9,20 +12,37 @@ namespace IntegrationWithPharmacies.Controllers
...
+        private List<Message> getAllForVue()
+        {
+            List<Message> messageForVue = new List<Message>();
+            foreach(Message message in MessageService.GetAll())
             {
-                return Ok(MessageService.GetAll());
+                RegistrationInPharmacy registration = RegistrationInPharmacyService.getPharmacyName(GetName(message));
+                if (registration != null) { messageForVue.Add(message); }
             }
+            return messageForVue;
```
Comment (trki293) [inline IntegrationWithPharmacies/Controllers/ActionsAndBenefitsController.cs:38]: -Refactor code. Replace method getAllForVue() with Where syntax, for example :
return MessageService.GetAll().Where(message => (RegistrationInPharmacyService.getPharmacyName(GetName(message))!=null) )
-Add a method that checks if the object is null and apply it in refactoring.

File: IntegrationWithPharmacies/Controllers/ActionsAndBenefitsController.cs [hunk 2]
```diff
@@ -9,20 +12,37 @@ namespace IntegrationWithPharmacies.Controllers
...
+        }
 
-
-
+        private static string GetName(Message message)
+        {
+            string[] parts = message.Text.Split(':');
+            return parts[0].Trim();
```
Comment (trki293) [inline IntegrationWithPharmacies/Controllers/ActionsAndBenefitsController.cs:44]: Write this in form:
"return message.Text.Split(':')[0].Trim()"

File: IntegrationWithPharmacies/Controllers/ReportController.cs [hunk 3]
```diff
@@ -32,7 +36,20 @@ public IActionResult Post(DateOfOrder date)
...
+            {
+                stringBuilder2.Append(registration.ApiKey + ";");
+
+            }
+            String str = stringBuilder2.ToString();
+            str = str + "!";
+            str = str + "             Report about consumption of medicine\n\n\n";
+
+            str = str +getReportText(date);
+            System.IO.File.WriteAllText(testFile, str);
```
Comment (trki293) [inline IntegrationWithPharmacies/Controllers/ReportController.cs:52]: Write this in form:
"System.IO.File.WriteAllText(testFile, stringBuilder2.ToString()+"!             Report about consumption of medicine\n\n\ngetReportText(date)");"

File: IntegrationWithPharmacies/Controllers/ReportController.cs [hunk 4]
```diff
@@ -32,7 +36,20 @@ public IActionResult Post(DateOfOrder date)
         {
             var sftpService = new SftpService(new NullLogger<SftpService>(), getConfig());
             var testFile = @"..\TextFile.txt";
-            System.IO.File.WriteAllText(testFile, getReportText(date));
+            StringBuilder stringBuilder2 = new StringBuilder();
+
+            List<RegistrationInPharmacy> registrations = RegistrationInPharmacyService.GetAll();
+            foreach (RegistrationInPharmacy registration in registrations)
```
Comment (trki293) [inline IntegrationWithPharmacies/Controllers/ReportController.cs:42]: Write this in form:
"foreach (RegistrationInPharmacy registration in RegistrationInPharmacyService.GetAll())"

File: IntegrationWithPharmacies/Controllers/ReportController.cs [hunk 5]
```diff
@@ -32,7 +36,20 @@ public IActionResult Post(DateOfOrder date)
         {
             var sftpService = new SftpService(new NullLogger<SftpService>(), getConfig());
             var testFile = @"..\TextFile.txt";
-            System.IO.File.WriteAllText(testFile, getReportText(date));
+            StringBuilder stringBuilder2 = new StringBuilder();
```
Comment (trki293) [inline IntegrationWithPharmacies/Controllers/ReportController.cs:39]: Change stringBuilder2 to stringBuilder.

File: IntegrationWithPharmacies/Program.cs [hunk 6]
```diff
@@ -13,7 +13,7 @@ public class Program
 
         public static void Main(string[] args)
         {
-            CreateHostBuilderMessages(args).Build();
+            //CreateHostBuilderMessages(args).Build().Run();
```
Comment (trki293) [inline IntegrationWithPharmacies/Program.cs:16]: Delete the commented line of code.

File: IntegrationWithPharmacies/RabbitMQService.cs [hunk 7]
```diff
@@ -21,37 +20,26 @@ public override Task StartAsync(CancellationToken cancellationToken)
...
                     message = JsonConvert.DeserializeObject<Message>(jsonMessage, new MyDateTimeConverter());
                 }
-
-                message.PharmacyName = "Apoteka Jankovic";
                 Program.ListOfMessages.Add(message);
-
+             
                 MessageService messageService = new MessageService();
-                messageService.Create(new MessageDto(message.Text, message.TimeStamp, "Apoteka Jankovic"));
+                //messageService.Create(new MessageDto(message.Text, message.TimeStamp, "Apoteka Jankovic", message.DateAction));
```
Comment (trki293) [inline IntegrationWithPharmacies/RabbitMQService.cs:39]: Delete the commented line of code.

File: PatientWebApplication/HealthClinic.DAL/Adapters/DoctorAdapter.cs [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+        }
+
+        public List<DoctorDto> ConvertDoctorListToDoctorDtoList(List<DoctorUser> doctors)
+        {
+            List<DoctorDto> doctorsDto = new List<DoctorDto>();
+            foreach (DoctorUser doctor in doctors)
+            {
+                doctorsDto.Add(DoctorToDoctorDto(doctor));
+            }
+            return doctorsDto;
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Adapters/DoctorAdapter.cs:23]: Refactor code. Replace method ConvertDoctorListToDoctorDtoList(List<DoctorUser> doctors) with Where syntax, for example :
return doctors.Where(doctor => UtilityMethods.CheckForSpecialty(doctor, specialty) )

File: PatientWebApplication/HealthClinic.DAL/Repository/MessageRepository.cs [hunk 9]
```diff
@@ -14,13 +11,12 @@ public class MessageRepository : IMessageRepository
 
         public MessageRepository()
         {
-            this.DbContext = new MyDbContext(new DbContextOptionsBuilder<MyDbContext>().UseMySql("Server=localhost;port=3306;Database=MYSQLHealtcareDB;user=root;password=root").UseLazyLoadingProxies().Options);
+            //DbContext = new MyDbContext(new DbContextOptionsBuilder<MyDbContext>().UseMySql("Server=localhost;port=3306;Database=MYSQLHealtcareDB;user=root;password=root").UseLazyLoadingProxies().Options);
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Repository/MessageRepository.cs:14]: Delete the commented line of code.

File: PatientWebApplication/HealthClinic.DAL/Repository/RegistrationInPharmacyRepository.cs [hunk 10]
```diff
@@ -29,7 +29,16 @@ public RegistrationInPharmacy getPharmacyApiKey(String apiKey)
         {
             return DbContext.Registrations.ToList().SingleOrDefault(registration => registration.ApiKey == apiKey);
         }
+        public Boolean Remove(String apiKey)
+        {
+            RegistrationInPharmacy registration2 = DbContext.Registrations.ToList().SingleOrDefault(registration => registration.ApiKey == apiKey);
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Repository/RegistrationInPharmacyRepository.cs:34]: Delete the indicated line of code.

File: PatientWebApplication/HealthClinic.DAL/Repository/RegistrationInPharmacyRepository.cs [hunk 11]
```diff
@@ -29,7 +29,16 @@ public RegistrationInPharmacy getPharmacyApiKey(String apiKey)
         {
             return DbContext.Registrations.ToList().SingleOrDefault(registration => registration.ApiKey == apiKey);
         }
+        public Boolean Remove(String apiKey)
+        {
+            RegistrationInPharmacy registration2 = DbContext.Registrations.ToList().SingleOrDefault(registration => registration.ApiKey == apiKey);
+            try {
+                DbContext.Registrations.Remove(registration2);
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Repository/RegistrationInPharmacyRepository.cs:36]: Write this in form:
"DbContext.Registrations.Remove( DbContext.Registrations.ToList().SingleOrDefault(registration => registration.ApiKey == apiKey));"

File: PatientWebApplication/HealthClinic.DAL/Service/RegistrationInPharmacyService.cs [hunk 12]
```diff
@@ -52,6 +52,14 @@ public RegistrationInPharmacy getPharmacyApiKey(String apiKey)
                 if (registration.ApiKey.Equals(apiKey))  return registration; 
             }
             return null;
+        }
+        public RegistrationInPharmacy getPharmacyName(String name)
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/RegistrationInPharmacyService.cs:56]: Rename the method to GetRegistrationByPharmacyName(String name).

File: PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs [hunk 13]
```diff
@@ -82,6 +85,12 @@ public List<DoctorAppointment> GetAppointmentsForPatient(int id)
             return _appointmentRepository.GetAppointmentsForPatient(id);
         }
 
+        public List<DoctorAppointment> GetAppointmentsForDoctor(int id)
+        {
+            List<DoctorAppointment> appointments = _appointmentRepository.GetAppointmentsForDoctor(id);
+            return appointments;
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs:91]: Write this in form:
"return _appointmentRepository.GetAppointmentsForDoctor(id)"

File: PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs [hunk 14]
```diff
@@ -92,17 +101,81 @@ public DoctorAppointment RecommendAnAppointment(DoctorUser doctor, DateTime date
...
+        {
+            List<DoctorAppointment> appointments = new List<DoctorAppointment>();
+            foreach (DoctorAppointment appointment in GetAppointmentsForDoctor(doctorId))
+            {
+                if(date == UtilityMethods.ParseDateInCorrectFormat(appointment.Date))
+                {
+                    appointments.Add(appointment);
+                }
+            }
+            return appointments;
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs:114]: Refactor code. Replace method etAllAppointmentsByDateAndDoctor(DateTime date, int doctorId) with Where syntax, for example :
return GetAppointmentsForDoctor(doctorId).Where(appointment  =>(date == UtilityMethods.ParseDateInCorrectFormat(appointment.Date)) );

File: PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs [hunk 15]
```diff
@@ -92,17 +101,81 @@ public DoctorAppointment RecommendAnAppointment(DoctorUser doctor, DateTime date
...
+        {
+            List<DoctorAppointment> appointments = new List<DoctorAppointment>();
+            foreach (DoctorAppointment appointment in GetAppointmentsForPatient(patientId))
+            {
+                if (date == UtilityMethods.ParseDateInCorrectFormat(appointment.Date))
+                {
+                    appointments.Add(appointment);
+                }
+            }
+            return appointments;
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs:127]: Refactor code. Replace method GetAllAppointmentsByDateAndPatient(DateTime date, int patientId) with Where syntax, for example :
return GetAppointmentsForPatient(patientId).Where(appointment =>(date == UtilityMethods.ParseDateInCorrectFormat(appointment.Date)) );

**Review by trki293 — CHANGES_REQUESTED** [review]
Correct the indicated places in the code.

**Review by trki293 — APPROVED** [review]
PR is approved.

**Review by Maja0505 — APPROVED** [review]
PR is approved. Good practices and principles of clean code are followed.

**Pull request merged**